src/rules/no-non-numeric-dimensions/__tests__/index.js

```diff
@@ -0,0 +1,54 @@
+import rule, { ruleName, messages, units } from "..";
+
+testRule(rule, {
+  ruleName,
+  config: [true],
+  syntax: "scss",
+  accept: loopOverUnits({
+    code: `
+    p {
+      padding: "1" * 1%unit%;
```

Comment:
```suggestion
      padding: 1 * 1%unit%;
```

...otherwise you're multiplying a string by a number which won't work.